### PR TITLE
Add new feature : TemporaryFile class

### PR DESCRIPTION
**Pull request by Nek-**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,3 +151,74 @@ DateTimeComparator::lowest($dateTime1, $dateTime2) : ?\DateTimeInterface
...
 * `$dateTime2` The second \DateTimeInterface or null
+
+### Temporary file management
+
+The class `TemporaryFile` helps you to create temporary file with ease.
+
+#### ::__construct()
+
+```php
+TemporaryFile::__construct(TemporaryDirectory $dir = null, string $path = '')
```
Comment (Awkan) [inline README.md]: TemporaryFile::__construct(TemporaryDirectory $dir = null, string $prefix = '')

File: src/Utils/Tempfile/TemporaryFile.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+use Nekland\Utils\Exception\RuntimeException;
+use Nekland\Utils\Tempfile\Exception\CannotCreateFileException;
+use Nekland\Utils\Tempfile\Exception\ImpossibleToUpdateFileException;
+use Nekland\Utils\Tempfile\Exception\TemporaryFileAlreadyRemovedException;
+
+class TemporaryFile
+{
+    /**
+     * Tmp dir path. Do not use this var directly, uses `getTmpDirectory`.
```
Comment (Awkan) [inline src/Utils/Tempfile/TemporaryFile.php]: `Tmp dir path. Do not use this var directly, use `getTmpDirectory` instead.`

Is this comment mandatory ? The property is private ?

Comment (Nek-) [reply src/Utils/Tempfile/TemporaryFile.php]: Hum. This was an indication for a potential contributor. I can stay here I guess.

File: README.md (hunk 1 above)
Comment (Nek-) [reply README.md]: Fixed :)

File: src/Utils/Tempfile/TemporaryFile.php [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    {
+        return $this->removed;
+    }
+
+    /**
+     * @return string
+     */
+    public function getContent()
+    {
+        $content = \file_get_contents($this->file);
```
Comment (Valouleloup) [inline src/Utils/Tempfile/TemporaryFile.php]: Why don't you check with the ` $this->removed ` first here ?

Comment (Nek-) [reply src/Utils/Tempfile/TemporaryFile.php]: It could be, in order to throw an exception. https://github.com/Nekland/Tools/issues/11

**Pull request merged**